### PR TITLE
NMS-10228:  use user defined time zone for date formatting

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA: https://issues.opennms.org/browse/NMS-10228

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

**Comment by mvrueden** [comment]
Merge conflicts (-:

**Comment by patrick-schweizer** [comment]
@mvrueden conflicts are solved...

File: opennms-webapp/src/main/webapp/admin/userGroupView/users/modifyUser.jsp [hunk 1]
```diff
@@ -450,6 +453,29 @@
             <input class="form-control" type="text" id="homePhone" name="homePhone" value='<%=(homePhone == null ? "":homePhone)%>'/>
           </div>
         </div>
+
+    <div class="form-group">
+          <label for="timeZoneId" class="col-sm-2 control-label">Time Zone:</label>
```
Comment (j-white) [inline opennms-webapp/src/main/webapp/admin/userGroupView/users/modifyUser.jsp]: Can we move this under a new `<h3>` tag? Right now it looks like it's part of the `Notification Information`.

Maybe add a new section called `Date/Time Preferences`?

File: opennms-webapp/src/main/webapp/admin/userGroupView/users/modifyUser.jsp [hunk 2]
```diff
@@ -450,6 +453,29 @@
...
+          <div class="col-sm-10">
+            <select class="form-control" id="timeZoneId" name="timeZoneId">
+<%
+              List<String> zones = new ArrayList<>();
+              zones.add("");
```
Comment (j-white) [inline opennms-webapp/src/main/webapp/admin/userGroupView/users/modifyUser.jsp]: I was unable to switch back to using the default time zone after selecting a specific time zone. Should this work currently?

**Comment by j-white** [comment]
Thanks for the update, works well now.

**Pull request merged**